### ISSUE TITLE
Getting errors when extracting a dcc asset on a collaboration host which was imported from another machinery project by a collaboration client

### ISSUE THREAD

**Issue by Tisten** [OurMachinery/themachinery-public#984]
- **Version:** Latest Binary Release
- **OS:** Windows

When importing a .dcc_asset on a collaboration client, and trying to extract it on the host I don't get any extracted assets.

I recorded a video, and compressed that video with 7z. Since GitHub doesn't accept .7z files I renamed it to .zip. So please download it, rename it back to .7z and uncompress to see it:
[Desktop 2022.01.28 - 13.29.42.07.mp4.zip](https://github.com/OurMachinery/themachinery-public/files/7958656/Desktop.2022.01.28.-.13.29.42.07.mp4.zip)

I get these errors in the log:
```
❌ C:\Users\tiste\agent\_work\4\s\3rdParty\themachinery\foundation\the_truth.c(4591): error: Unexpected nil item (2)
❌ generated-code:211:18: error: use of undeclared identifier 'a'
    output.res = a * b;
                 ^
generated-code:279:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^
generated-code:293:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^
 (30)
❌ generated-code:496:18: error: use of undeclared identifier 'a'
    output.res = a * b;
                 ^
generated-code:564:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^
generated-code:578:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^

❌ generated-code:211:18: error: use of undeclared identifier 'a'
    output.res = a * b;
                 ^
generated-code:279:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^
generated-code:293:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^

❌ generated-code:496:18: error: use of undeclared identifier 'a'
    output.res = a * b;
                 ^
generated-code:564:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^
generated-code:578:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^
 (3)
❌ generated-code:211:18: error: use of undeclared identifier 'a'
    output.res = a * b;
                 ^
generated-code:279:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^
generated-code:293:22: error: use of undeclared identifier 'b'
    output.res = a * b;
                     ^
 (9)
```

**Comment by Tisten** [comment]
I have a few other issues to report about this scenario as well:
After importing a dcc_asset on a collaboration client and right away extracting them (withing a second of importing it), I get an untextured model on the client, and it doesn't get extracted at all on the host. If I at a later point then click on a .creation asset on the client (which doesn't exist on the server), the client crashes.

After importing a dcc_asset, if I wait a while and then extract the assets on the client, the asset will appear on the host. If I quickly click on a .creation asset on the host, then the host crashes with this dialogue:
![image](https://user-images.githubusercontent.com/6726075/151551234-c254d4af-cf36-48fb-b4f8-bb5fd9b558cd.png)
In this case the client will then also get the logs from above.

I'm sometimes also getting:
❌ c:\work\themachinery\foundation\the_truth.c(4633): error: Unexpected nil item

I am not always getting crash dumps for my crashes, but I've at least gotten these two, using the 2022.1 release.
[crashdumps.zip](https://github.com/OurMachinery/themachinery-public/files/7958830/crashdumps.zip)

**Comment by Tisten** [comment]
One more:
[the_machinery_exe 2022-01-28 14.10.30.dmp.zip](https://github.com/OurMachinery/themachinery-public/files/7958848/the_machinery_exe.2022-01-28.14.10.30.dmp.zip)

**Comment by niklas-ourmachinery** [comment]
I think the fix to #983 fixed some of these issues -- extraction on the host seems to work and I don't get the crash any more.

I'm not sure if all the timing related issues are fixed though... can you test and report a new issue if they are still there?
